src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -930,11 +950,10 @@ internal StackFrameVersionOne CreateStackFrame(StackFrame v2StackFrame)
                     string fqln = location.FullyQualifiedLogicalName;
 
                     if (_currentV2Run.LogicalLocations != null &&
-                        _currentV2Run.LogicalLocations.ContainsKey(fqln) &&
-                        !string.IsNullOrWhiteSpace(_currentV2Run.LogicalLocations[fqln].FullyQualifiedName))
+                        !string.IsNullOrWhiteSpace(_currentV2Run.LogicalLocations[location.LogicalLocationIndex].FullyQualifiedName))
```

Comment:
>LogicalLocationIndex [](start = 91, length = 20)

Test for -1. DRY out code by computing this expression only once: `_currentV2Run.LogicalLocations[location.LogicalLocationIndex].FullyQualifiedName` (and as usual, it's FQLN ?? Name.

Reply:
resolve as part of [https://github.com/Microsoft/sarif-sdk/issues/1181](https://github.com/Microsoft/sarif-sdk/issues/1181)

---
In reply to: [243867281](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243867281) [](ancestors = 243867281)

Reply:
Reactivated for the sake of adding the -1 check (otherwise I'll forget). The rest of the change is covered by #1181.

---
In reply to: [244054845](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r244054845) [](ancestors = 244054845,243867281)